test/api-tests.js

```diff
@@ -1,52 +1,138 @@
+const serviceRoot = $("#service-root").attr("value");
 
-describe("Uniprot API", function () {
-    var request;
 
-    beforeEach(function () {
-        window.$ = $;
-        request = sinon.useFakeXMLHttpRequest();
-    });
 
-    afterEach(function () {
-        request.restore();
-    });
 
-    it("should load", function () {
-        geneFunctions.getUniProtInfo(gene).done(function (result) {
-            expect(result).to.not.be.undefined;
-        });
-    });
-});
+const {XMLSerializer} = require("w3c-xmlserializer");
+global.XMLSerializer = XMLSerializer.expose.Window.XMLSerializer;
 
+const apis = require(__dirname + "/../web-client/public/gene/api.js");
 
-describe("Ensembl API", function () {
-    it("should load", function () {
-    });
-});
+let uniprotDoc = document.implementation.createDocument("", "", null);
+
+let uniprot = uniprotDoc.createElement("uniprot");
+uniprotDoc.appendChild(uniprot);
+
+let entry = uniprotDoc.createElement("entry");
+uniprot.appendChild(entry);
+
+let name = uniprotDoc.createElement("name");
+let sequence = uniprotDoc.createElement("sequence");
+entry.appendChild(name);
+entry.appendChild(sequence);
+
+let nameText = uniprotDoc.createTextNode("YHP1_YEAST");
+let sequenceText = uniprotDoc.createTextNode("MESRNTVLPSLPNIITGTSNSPFQLHTLPNTNFPSDDQGDIRLPPLAASAHIVRPVVNIY" +
+"KSPCDEERPKRKSPQAVDFLSQRVTTSMTPLSKPKKLSSHSPFTPTVRVCSKEQPPQSMH" +
+"SYKKVNILTPLSAAKAVLTPTTRKEKKRSFAFITHSQETFPKKEPKIDNARLARRKRRRT" +
+"SSYELGILQTAFDECPTPNKAKRIELSEQCNMSEKSVQIWFQNKRQAAKKHKNSGNTSHC" +
+"KVHSNDSMSMISYSDAALEITSTPTSTKEAITAELLKTSPANTSSIFEDHHITPCKPGGQ" +
+"LKFHRKSVLVKRTLSNTGHSEIIKSPKGKENRLKFNAYERKPLGEVDLNSFKN");
+name.appendChild(nameText);
+sequence.appendChild(sequenceText);
+
+let protein = uniprotDoc.createElement("protein");
+entry.appendChild(protein);
+
+let reccomendedName = uniprotDoc.createElement("reccomendedName");
+protein.appendChild(reccomendedName);
 
-describe("JASPAR API", function () {
-    it("should load", function () {
+let fullName = uniprotDoc.createElement("fullName");
+reccomendedName.appendChild(fullName);
+
+let fullNameText = uniprotDoc.createTextNode("Homeobox protein YHP1");
+fullName.appendChild(fullNameText);
+
+let organism = uniprotDoc.createElement("organism");
+entry.appendChild(organism);
+
+let organismName = uniprotDoc.createElement("name");
+organism.appendChild(organismName);
+
+let organismNameText = uniprotDoc.createTextNode("Saccharomyces cerevisiae (strain ATCC 204508 / S288c)");
+organismName.appendChild(organismNameText);
+
+
+ const gene = "YHP1";
+  describe("getUniProtInfo", () => {
+    let stub;
+    beforeEach(() => {
+        stub = sinon.stub(global.window.api, "getUniProtInfo");
     });
-});
+    afterEach(() => {
+        stub.restore();
+    });
+    it("should display the correct data", done => {
+        stub.returns(Promise.resolve(uniprotDoc));
+        global.window.api.getUniProtInfo(gene).then(info => {
+          console.log(info);
```

Comment:
Let's try to obscure/hide this from the specific PR because otherwise it blurs the focus of the PR. You have this in git already so it can be safely removed then restored later.